bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -1274,6 +1312,31 @@ void ReplicaImp::onInternalMsg(InternalMessage &&msg) {
   if (auto *ppm = std::get_if<PrePrepareMsg *>(&msg)) {
     return startConsensusProcess(*ppm, true);
   }
+
+  // Handle a the PrePrepare Carrier
+  if (auto *ppcim = std::get_if<PrePrepareCarrierInternalMsg>(&msg)) {
+    if (bftEngine::ControlStateManager::instance().getPruningProcessStatus()) {
```

Comment:
Here also, you cannot handle any message while state transfer. You should reject and delete the message if the replica is collecting state
e.g.
```
if(!isCollectingState())
```

Reply:
Thanks @yontyon , yes I am now keeping the isCollectingState() check all the places before the onMessage() function call.